### PR TITLE
readable-stream v3, nodejs v16

### PR DESCRIPTION
**Pull request by legobeat**
- **BREAKING:** Increase minimum Node.js version from 12 to 16
  - ci: Run on Node.js v18,20 instead of 12,14
  - Change recommended Node.js version to 18
- **BREAKING:** deps: `readable-stream`@`^2.3.3`->`^3.6.2`
  - deps: replace `end-of-stream` with port in `readable-stream`
  - devDeps: replace `pump` with port in `readable-stream`

### CODE REVIEW

File: src/readable-stream.d.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// import indicrectly used for module augmentation to extend @types/readable-stream with missing exports
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import * as readableStream from 'readable-stream';
```
Comment (naugtur) [inline src/readable-stream.d.ts:3]: Why is that not part of the original module's types?

File: yarn.lock [hunk 2]
```diff
@@ -126,9 +126,9 @@
     read-package-json-fast "^2.0.1"
 
 "@types/end-of-stream@^1.4.0":
```
Comment (naugtur) [inline yarn.lock:128]: (nit) these should no longer be necessary

File: test/index.js [hunk 3]
```diff
@@ -120,6 +123,17 @@ function basicTestSetup() {
...
-  stream.on('data', (chunk) => results.push(chunk));
+  let flushed = false;
+  function onFinish(err) {
+    if (flushed) {
+      return;
+    }
+    flushed = true;
+    callback(err, results);
+  }
+  stream.prependListener('close', onFinish);
```
Comment (kumavis) [inline test/index.js:134]: is this redundant?

File: src/readable-stream.d.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+// import indicrectly used for module augmentation to extend @types/readable-stream with missing exports
```
Comment (kumavis) [inline src/readable-stream.d.ts:1]: sp `indirectly`

**Review by kumavis — APPROVED** [review]

**Pull request merged**